src/Microsoft.Data.SqlClient/tests/FunctionalTests/AlwaysEncryptedTests/SqlColumnEncryptionCertificateStoreProviderShould.cs

```diff
@@ -225,6 +226,12 @@ public void EncryptAndDecryptDataSuccessfully()
         [PlatformSpecific(TestPlatforms.Windows)]
         public void TestCEKEncryptionReversal(StoreLocation certificateStoreLocation, String certificateStoreNameAndLocation)
         {
+            // If LocalMachine cert path is needed, but current role is not Admin, skip the test.
+            if (StoreLocation.LocalMachine == certificateStoreLocation && !CertificateFixture.IsAdmin)
+            {
+                return;
+            }
+
```

Comment:
I don't think this is good way of skipping the test, the test will still pass, even if it didn't run at all.
Please create a Conditional Fact and add it to the tests where this is needed.

Reply:
I don't think it is a good way either, the thing is that I don't know if there's a good way to make a condition check on the inline data. Conditional Fact or it is actually Conditional theory in this case won't work as expected. I'm thinking to refactor the inline data into a DataAttribute, and add condition in the return list to get rid of this line.

Reply:
As long as we don't say the test passed in the end, and that the test skipped due to missing permissions, that is ideal solution here.